nodebuilder/fraud/service.go

```diff
@@ -2,21 +2,77 @@ package fraud
 
 import (
 	"context"
+	"encoding/json"
 
 	"github.com/celestiaorg/celestia-node/fraud"
 )
 
-// Module encompasses the behavior necessary to subscribe and broadcast fraud proofs within the network.
-// Any method signature changed here needs to also be changed in the API struct.
-type Module interface {
+var _ Module = (*Service)(nil)
+
+// Service is an implementation of Module that uses fraud.Service as a backend. It is used to provide fraud proofs as a
+// non-interface type to the API, and wrap fraud.Subscriber with a channel of Proofs.
+type Service struct {
 	fraud.Service
 }
 
-// API is a wrapper around Module for the RPC.
-// TODO(@distractedm1nd): These structs need to be autogenerated.
-//
-//go:generate go run github.com/golang/mock/mockgen -destination=mocks/api.go -package=mocks . Module
-type API struct {
-	Subscribe func(fraud.ProofType) (fraud.Subscription, error)
-	Get       func(context.Context, fraud.ProofType) ([]fraud.Proof, error)
+func (s *Service) Subscribe(ctx context.Context, proofType fraud.ProofType) (chan Proof, error) {
+	subscription, err := s.Service.Subscribe(proofType)
+	if err != nil {
+		return nil, err
+	}
+	concreteProofs := make(chan Proof)
+	go func() {
+		proof, _ := subscription.Proof(ctx)
+		concreteProofs <- Proof{Proof: proof}
+	}()
+	return concreteProofs, nil
+}
+
+func (s *Service) Get(ctx context.Context, proofType fraud.ProofType) ([]Proof, error) {
+	proofs, err := s.Service.Get(ctx, proofType)
+	if err != nil {
+		return nil, err
+	}
+	concreteProofs := make([]Proof, len(proofs))
+	for i, proof := range proofs {
+		concreteProofs[i].Proof = proof
+	}
+	return concreteProofs, nil
+}
+
+// Proof embeds the fraud.Proof interface type to provide a concrete type for JSON serialization.
+type Proof struct {
```

Comment:
Try without embedding

Reply:
doesn't work; because the types we are embedding are interfaces, and the types we are defining need to be structs

Reply:
resolved?

Reply:
Yes